src/archive.rs

```diff
@@ -1,79 +1,90 @@
 use actix_web::http::ContentEncoding;
-use bytes::Bytes;
 use libflate::gzip::Encoder;
 use serde::Deserialize;
-use std::io;
 use std::path::Path;
 use strum_macros::{Display, EnumIter, EnumString};
 use tar::Builder;
 
 use crate::errors::ContextualError;
 
 /// Available compression methods
-#[derive(Deserialize, Clone, EnumIter, EnumString, Display)]
+#[derive(Deserialize, Clone, Copy, EnumIter, EnumString, Display)]
 #[serde(rename_all = "snake_case")]
 #[strum(serialize_all = "snake_case")]
 pub enum CompressionMethod {
-    /// TAR GZ
+    /// Gzipped tarball
     TarGz,
+
+    /// Regular tarball
+    Tar,
 }
 
 impl CompressionMethod {
-    pub fn extension(&self) -> String {
-        match &self {
+    pub fn extension(self) -> String {
+        match self {
             CompressionMethod::TarGz => "tar.gz",
+            CompressionMethod::Tar => "tar",
         }
         .to_string()
     }
 
-    pub fn content_type(&self) -> String {
-        match &self {
+    pub fn content_type(self) -> String {
+        match self {
             CompressionMethod::TarGz => "application/gzip",
+            CompressionMethod::Tar => "application/tar",
         }
         .to_string()
     }
 
-    pub fn content_encoding(&self) -> ContentEncoding {
-        match &self {
+    pub fn content_encoding(self) -> ContentEncoding {
+        match self {
             CompressionMethod::TarGz => ContentEncoding::Gzip,
+            CompressionMethod::Tar => ContentEncoding::Identity,
         }
     }
-    /// Creates an archive of a folder, using the algorithm the user chose from the web interface
-    /// This method returns the archive as a stream of bytes
-    pub fn create_archive<T: AsRef<Path>>(
-        &self,
+
+    pub fn create_archive<T, W>(
```

Comment:
Since this is rather abstract and complex now, I'd definitely like some documentation here.

Reply:
Added documentation for each step - though there's not _that much_ difference with the previous functions.